libtransact/src/context/manager/thread/mod.rs

```diff
@@ -295,3 +295,169 @@ impl ContextManagerThread {
             .expect("Could not build a thread for Context Manager")
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::collections::HashMap;
+    use std::thread::JoinHandle;
+
+    use crate::context::ContextExecution;
+    use crate::protocol::receipt::{Event, EventBuilder, StateChange};
+    use crate::state::hashmap::HashMapState;
+
+    static KEY1: &str = "111111111111111111111111111111111111111111111111111111111111111111";
+    static KEY2: &str = "222222222222222222222222222222222222222222222222222222222222222222";
+    static KEY3: &str = "333333333333333333333333333333333333333333333333333333333333333333";
+    static KEY4: &str = "444444444444444444444444444444444444444444444444444444444444444444";
+
+    static BYTES1: [u8; 4] = [0x01, 0x02, 0x03, 0x04];
+    static BYTES2: [u8; 4] = [0x05, 0x06, 0x07, 0x08];
+    static BYTES3: [u8; 4] = [0x09, 0x10, 0x11, 0x12];
+
+    static EVENT_TYPE1: &str = "sawtooth/block-commit";
+    static ATTR1: (&str, &str) = (
+        "block_id",
+        "f40b90d06b4a9074af2ab09e0187223da7466be75ec0f472 \
+         f2edd5f22960d76e402e6c07c90b7816374891d698310dd25d9b88dce7dbcba8219d9f7c9cae1861",
+    );
+    static ATTR2: (&str, &str) = ("block_num", "3");
+
+    fn shutdown_thread(join_handle: JoinHandle<()>) {
+        thread::Builder::new()
```

Comment:
This should not spawn a new thread, but just join the thread directly.

It also doesn't actually signal for the context manager thread to shut down, so if you remove it, the test hangs.

Reply:
The tests hang if this function just joins directly, hence spawning a new thread. This pertains to one of the questions I had for this PR, in how to handle shutting down this thread. Wasn't quite sure what best practice would be in this case.

Reply:
It should signal the background thread using the `Shutdown` enum value